ibis/expr/types/generic.py

```diff
@@ -2483,7 +2482,7 @@ class NullValue(Value):
 
 
 @public
-class NullScalar(Scalar, NullValue, Singleton):
+class NullScalar(Scalar, NullValue):
```

Comment:
`Expr` is not a subclass of `Annotable` so `singleton=True` cannot be used here, although we could turn `Expr` into `Annotable` as well, should benchmark the impact of that.